### PR TITLE
Add worker to post vendor integration reports to Slack channels

### PR DESCRIPTION
**Pull request by mnacos**
## Context

Basically renders the stats obtained by `VendorAPIMonitor` as pre-formatted text and posts the report to an appropriate Slack channel.

This is a first PR to ship the report-generation code, which we can then test on various environments, including production, by triggering the worker manually. The worker is currently configured to send to a test Slack channel. Once we are happy with the formatting and the data, we can reconfigure the worker to use the real channels and schedule its runs using `clock` in a follow-up PR.

## Changes proposed in this pull request

This is the kind of slack message that gets sent:

image

## Guidance to review

Requires vendor-specific environment variables to be present, e.g. `TRIBAL_INTEGRATION_STATS_SLACK_URL`.

Whether a report is actually sent for a vendor depends on both the presence of the relevant environment variable and on whether we've scheduled any such jobs. We could also choose to schedule reports for some providers to private channels, for internal use.

## Link to Trello card

https://trello.com/c/Mxtr0D9q

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/workers/vendor_integration_stats_worker.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    if @webhook_url.present?
+      message = SlackReport.new(vendor_name).generate
+      post_to_slack message
+    end
+  end
+
+private
+
+  def fetch_webhook_url
+    ENV.fetch "#{@vendor_name.upcase}_INTEGRATION_STATS_SLACK_URL"
```
Comment (duncanjbrown) [inline app/workers/vendor_integration_stats_worker.rb:24]: Why do we have to use separate URLs for each vendor?

Comment (mnacos) [reply app/workers/vendor_integration_stats_worker.rb:24]: Each url is tied to a different Slack channel

Comment (duncanjbrown) [reply app/workers/vendor_integration_stats_worker.rb:24]: ach — they used to be more permissive!

File: app/workers/vendor_integration_stats_worker.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
+require 'http'
+
+class VendorIntegrationStatsWorker
+  include Sidekiq::Worker
+
+  SLACK_CHANNELS = {
+    'tribal' => 'tribal_dfe_collaboration_',
+    'ellucian' => 'ellucian_dfe_collaboration',
```
Comment (cwrw) [inline app/workers/vendor_integration_stats_worker.rb:8]: could we add the other two channels?

File: app/workers/vendor_integration_stats_worker.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    response = HTTP.post(@webhook_url, body: payload.to_json)
+
+    unless response.status.success?
+      raise SlackMessageError, "Slack error: #{response.body}"
+    end
+  end
+
+  class SlackMessageError < StandardError; end
+
+  class SlackReport
```
Comment (cwrw) [inline app/workers/vendor_integration_stats_worker.rb:46]: could we move this into it's own service?

**Review by cwrw — APPROVED** [review]
looking good, some minor comments

File: app/workers/vendor_integration_stats_worker.rb (hunk 2 above)
Comment (mnacos) [reply app/workers/vendor_integration_stats_worker.rb:8]: done

File: app/workers/vendor_integration_stats_worker.rb (hunk 3 above)
Comment (mnacos) [reply app/workers/vendor_integration_stats_worker.rb:48]: Leaving this as is for now, as nothing else uses it.

**Pull request merged**